### ISSUE TITLE
Softmatch implementation discrepancy

### ISSUE THREAD

**Issue by Alotra** [microsoft/Semi-supervised-learning#89]
Hello, 

First, thanks for the great work that has been done here.

Then, I have a question on the SoftMatch implementation. After having uniformly aligned the pseudo labels generated on the weakly augmented unlabeled data (line **87** of _softmatch.py_), you re-use those uniformly aligned pseudo labels for the _update()_ function of the **SoftMatchWeightingHook(MaskingHook)** to compute the batch mean and the batch var that are used in the EMA update of the global mean and var (here called _'self.prob_max_mu_t_' and '_self.prob_max_var_t_').
But in the SoftMatch paper, and in the TorchSSL implementation of Softmatch, I think that this update is done before aligning the pseudo labels, so with the softmax logits of the model.
Is it an update of the algorithm ?

Thanks in advance.

**Comment by Hhhhhhao** [comment]
Hi,

Good catch. There is no update of the algorithm. In the paper, we use torchssl implementation for Table.1 results (and ImageNet results). The NLP results is from USB. This is probably why on classic setting in USB, SoftMatch performs lower than classic setting in USB (but the difference is trivial).
